src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,97 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private static final double INCREMENT_CAPACITY = 1.5;
+    private int size;
+    private T[] elements;
+    private int arrayCapacity;
+
+    public ArrayList() {
+        arrayCapacity = INITIAL_CAPACITY;
+        elements = (T[]) new Object[arrayCapacity];
+    }
+
+    private void grow() {
+        arrayCapacity *= INCREMENT_CAPACITY;
+        T[] tempArray = (T[]) new Object[arrayCapacity];
+        System.arraycopy(elements, 0, tempArray, 0, size);
+        elements = tempArray;
+    }
+
+    private void checkItem(int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index is invalid");
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elements.length) {
+            grow();
+        }
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkItem(index);
+        if (size == elements.length) {
+            grow();
+        }
```

Comment:
do we need this in this method?

Reply:
Yes, we need grow().
I gaves us prrortunity to increse array length.

Reply:
I can moved it inside method, but it increse amount of line.

Reply:
You may mean I had to delete it in a previous add(T value).